File: SDKs/amazonaws.com/codestar/2017-04-19/go/pkg/models/operations/listprojects.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ListProjectsXAmzTargetEnum string

const (
	ListProjectsXAmzTargetEnumCodeStar20170419ListProjects ListProjectsXAmzTargetEnum = "CodeStar_20170419.ListProjects"
)

type ListProjectsHeaders struct {
	XAmzAlgorithm     *string                    `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                    `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                    `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                    `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                    `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                    `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                    `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        ListProjectsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type ListProjectsRequest struct {
	Headers ListProjectsHeaders
	Request shared.ListProjectsRequest `request:"mediaType=application/json"`
}

type ListProjectsResponse struct {
	ContentType               string
	InvalidNextTokenException *interface{}
	ListProjectsResult        *shared.ListProjectsResult
	StatusCode                int64
	ValidationException       *interface{}
}

File: SDKs/amazonaws.com/codestar/2017-04-19/typescript/src/sdk/models/operations/listprojects.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum ListProjectsXAmzTargetEnum {
    CodeStar20170419ListProjects = "CodeStar_20170419.ListProjects"
}


export class ListProjectsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: ListProjectsXAmzTargetEnum;
}


export class ListProjectsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: ListProjectsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ListProjectsRequest;
}


export class ListProjectsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  invalidNextTokenException?: any;

  @SpeakeasyMetadata()
  listProjectsResult?: shared.ListProjectsResult;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  validationException?: any;
}
